### PR TITLE
Switch Delay

### PR DESCRIPTION
**Pull request by SecretAgent12**
Video - https://youtu.be/sg47rhzMxn8

### CODE REVIEW

**Comment by SecretAgent12** [comment]
i forgot to attach the video

**Comment by GoldSloth** [comment]
How does this work when switching to other guns?

**Comment by SecretAgent12** [comment]
similar

File: src/main/java/com/flansmod/common/guns/ItemGun.java [hunk 1]
```diff
@@ -282,6 +282,30 @@ public void addInformation(ItemStack stack, EntityPlayer player, List lines, boo
...
+                            .max(PlayerHandler
+                                            .getPlayerData(Minecraft.getMinecraft().thePlayer, Side.CLIENT).shootTimeRight,
+                                    animationLength);
+                }
+
+            }
+        }
+
         if (entity instanceof EntityPlayer && ((EntityPlayer) entity).inventory.getCurrentItem() == itemstack) {
             //Get useful objects
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/ItemGun.java]: You can move all this code down to here - where the checks on the player are happening, already

File: src/main/java/com/flansmod/common/guns/ItemGun.java [hunk 2]
```diff
@@ -374,6 +398,14 @@ private void onUpdateClient(ItemStack itemstack, World world, Entity entity, int
 
                 //--------------------------------- Main hand item ---------------------------------------------
                 if (type.usableByPlayers) {
+                    GunAnimations animations = FlansModClient.getGunAnimations(player, false);
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/ItemGun.java]: This line looks redundant

File: src/main/java/com/flansmod/common/guns/ItemGun.java [hunk 3]
```diff
@@ -374,6 +398,14 @@ private void onUpdateClient(ItemStack itemstack, World world, Entity entity, int
...
+//                    animations.lookAt = LookAtState.NONE;
+                    float shootTime = data.shootTimeRight;
+
+                    // For each
+                    while (shootTime <= 0.0f) {
+                        // Add the delay for this shot and shoot it!
+                        shootTime += type.shootDelay;
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/ItemGun.java]: Where do you use shootTime?

**Comment by SecretAgent12** [comment]
corrected

**Comment by GoldSloth** [comment]
Looks good, thanks

**Pull request merged**